# ARETE_PLASMIDS_PAPER

## Overview

This repo contains the material and experiments for the plasmids paper of the ARETE grant.

The purpose of the paper is to analyze a large dataset of *E. faecium* samples for which hybrid short-reads and long-reads assemblies will be proevided.  

The short-reads data were generated in 2017 using Illumina while the long-reads data were generated in 2021 usaing Oxford Nanopore.

The hybrid data and short-reads data were assembled using Unicycler. The hybrid data, that will be used as a proxy for ground truth, were polished using Pilon.

The initial purpose of the work is to assess and compare the performances of several plasmids prediction methods. Potential foillow-up steps will include the analysis of the predicted plasmids.

## Data

**TO DO**: provide an overview of the data and metadata.

## Data preparation

#### Hybrid assemblies

The hybrid assemblies will be analyzed to extract chromosomal and plasmid contigs. To do so, features of each contig will be recorded in the spreadsheet
<a href="https://docs.google.com/spreadsheets/d/1TYZfiw3Tucnz9zZE81R3uaRorE0WGbjU6eGbrT2gTYI">hybrid contigs properties</a>. One issue is that the infortmation about contigs circularity might be removed from the FASTA header by Pilon and we will need to check if Pilon modifies the length of the contigs after polishing; if not, then the feature can be extracted from the pre-polishing contigs from the FASTA header generated by Unicycler.